BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Antenna_Performance_Metric/rootAnalysis.py

```diff
@@ -1,297 +1,238 @@
 #!/usr/bin/env python
 #This file came from Will with some GENETIS specific modifications by Dylan
-import ROOT
-import numpy as np
-#import matplotlib.pyplot as plt
-#import scipy.signal
-from collections import defaultdict
+# Run with:
+# python rootAnalysis.py 1 1 19 /users/PAS1960/dylanwells1629/developing/GENETIS_PUEO/BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop/Generation_Data 2023_05_08 /users/PAS1960/dylanwells1629/developing/GENETIS_PUEO/BiconeEvolution/current_antenna_evo_build/XF_Loop/Evolutionary_Loop
+
+# Imports
 import os
-from fnmatch import fnmatch
-#import matplotlib
 import argparse
-parser = argparse.ArgumentParser()
+from collections import defaultdict
+from fnmatch import fnmatch
+
+import numpy as np
+
+import ROOT
+
+# Global Variables
+desc = 'Analyze the root files and output fitness scores and errors'
+parser = argparse.ArgumentParser(description = desc)
 parser.add_argument("gen", help="current gen", type=int)
 parser.add_argument("indiv", help="Individual", type=int)
 parser.add_argument("energy", help="Individual", type=int)
 parser.add_argument("opath", help="Output path", type=str)
 parser.add_argument("RunName", help="Run Name", type=str)
+parser.add_argument("WorkingDir", help="Working Directory", type=str)
 g=parser.parse_args()
 
+
+
 ROOT.gSystem.Load("/fs/ess/PAS1960/buildingPueoSim/pueoBuilder/lib/libNiceMC.so")
 ROOT.gSystem.Load("/fs/ess/PAS1960/buildingPueoSim/pueoBuilder/lib/libAntarcticaRoot.so")
 ROOT.gSystem.Load("/fs/ess/PAS1960/buildingPueoSim/pueoBuilder/lib/libAnitaEvent.so")
 ROOT.gSystem.Load("/fs/ess/PAS1960/buildingPueoSim/pueoBuilder/lib/libPueoSim.so")
```

Comment:
```suggestion
lib_dir = '/fs/ess/PAS1960/buildingPueoSim/pueoBuilder/lib/'
# lib_dir = '/users/PAS1960/dylanwells1629/buildingPueoSim/pueoBuilder/lib/'
ROOT.gSystem.Load(lib_dir + 'libNiceMC.so')
ROOT.gSystem.Load(lib_dir + 'libAntarcticaRoot.so')
ROOT.gSystem.Load(lib_dir + 'libAnitaEvent.so')
ROOT.gSystem.Load(lib_dir + 'libPueoSim.so')
```
This removes duplication and makes it easier to switch between dirs. 

You could also add `parser.add_argument("libdir", help=".so libraries directory", type=Path, default='/fs/ess/PAS1960/buildingPueoSim/pueoBuilder/lib/')` above and avoid hardcoding different dirs.